src/main/java/backend/Logic.java

```diff
@@ -231,9 +232,24 @@ public void removeUnusedModels(Set<String> reposInUse) {
                 .forEach(models::removeRepoModelById);
     }
 
-    public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
-        uiManager.updateRateLimits(rateLimits);
-        return rateLimits;
+    /**
+     * Updates the GitHub API quota info in the GUI.
+     * @param apiQuotaInfo The GitHub API quota information.
+     * @return the apiQuotaInfo instance
+     */
+    public ApiQuotaInfo updateApiQuotaInfoInGui(ApiQuotaInfo apiQuotaInfo) {
+        uiManager.updateApiQuotaInfo(apiQuotaInfo);
+        return apiQuotaInfo;
```

Comment:
Yeah, `updateSyncRefreshRate` passing the parameter is unavoidable, but you can change `updateApiQuotaInfoInGui` to return void.